pkg/traffic/gateway.go

```diff
@@ -122,3 +141,7 @@ func (a *Gateway) GetListenerByHost(host string) *gatewayv1beta1.Listener {
 func (a *Gateway) ExposesOwnController() bool {
 	return false
 }
+
+func (a *Gateway) IsMultiCluster() bool {
+	return a.isMultiCluster
```

Comment:
I'd probably avoid setting a boolean in the wrapper struct and only calculating it once in a constructor, but instead add the logic to  `IsMultiCluster` and do every time it's called.

Reply:
Presumably a false value from here could also mean "Unknown" since the Gateway might have no status yet?

Reply:
Why would you avoid it? Do we expect updated copies of Gateway CR mid-reconciliation, update `Status.Addresses[]`, or the boolean flag "manually" to invalidate the check? 
The idea I had: before doing any reconciliation we validate the Gateway we got and if it is fine we stop checking it singe Gateway should remain the same (any updates will be dealt with but a new loop with a new check in the beginning)

Reply:
If the gateway has no status - we don't really care about it being a single or multicluster one :thinking: